### PR TITLE
support MulNoNan for TF

### LINKED ISSUE #20758: [Good First Issue][TF FE]: Support MulNoNan operation for TensorFlow models
### Context

OpenVINO component responsible for support of TensorFlow models is called as TensorFlow Frontend (TF FE). TF FE converts a model represented in [TensorFlow opset](https://www.tensorflow.org/api_docs/python/tf/raw_ops) to a model in [OpenVINO opset](https://docs.openvino.ai/2023.1/openvino_docs_ops_opset.html).

In order to infer TensorFlow models with [MulNoNan](https://www.tensorflow.org/api_docs/python/tf/raw_ops/MulNoNan) operation by OpenVINO, TF FE needs to be extended with this operation support.

### What needs to be done?

For [MulNoNan](https://www.tensorflow.org/api_docs/python/tf/raw_ops/MulNoNan) operation support, you need to add the corresponding loader into TF FE [op](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/tensorflow_common/src/op) directory and to register it into the dictionary of [Loaders](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/src/op_table.cpp). One loader is responsible for conversion (or decomposition) of one type of TensorFlow operation.

Here is an example of loader implementation for TensorFlow `Einsum` operation:

```cpp
OutputVector translate_einsum_op(const NodeContext& node) { 
     auto op_type = node.get_op_type(); 
     TENSORFLOW_OP_VALIDATION(node, op_type == "Einsum", "Internal error: incorrect usage of translate_einsum_op."); 
     auto equation = node.get_attribute<std::string>("equation"); 
  
     OutputVector inputs; 
     for (size_t input_ind = 0; input_ind < node.get_input_size(); ++input_ind) { 
         inputs.push_back(node.get_input(input_ind)); 
     } 
  
     auto einsum = make_shared<Einsum>(inputs, equation); 
     set_node_name(node.get_name(), einsum); 
     return {einsum}; 
 } 
```

In this example, `translate_einsum_op` converts TF `Einsum` into OV `Einsum`. `NodeContext` object passed into the loader packs all info about inputs and attributes of `Einsum` operation. The loader retrieves an attribute of the equation by using the `NodeContext::get_attribute()` method, prepares input vector, creates `Einsum` operation from OV opset and returns a vector of outputs.

Responsibility of a loader is to parse operation attributes, prepare inputs and express TF operation via OV operations sub-graph. Example for `Einsum` demonstrates the resulted sub-graph with one operation. In PR https://github.com/openvinotoolkit/openvino/pull/19007 you can see operation decomposition into multiple node sub-graph.

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/19007

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [OpenVINO TensorFlow Frontend](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/README.md)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@openvinotoolkit/openvino-tf-frontend-maintainers 

### Ticket

TBD

### PR DESCRIPTION
**Pull request by ahmadchalhoub**
### Details:
Implementing support for the MulNoNan op in TensorFlow

### Tickets:
 - Closes #20758

### CODE REVIEW

**Comment by rkazants** [comment]
Hi @ahmadchalhoub, you did extra changes - I believe:) Please leave only ones for TF FE.

Best regards,
Roman

**Comment by ahmadchalhoub** [comment]
Hi @rkazants, thanks for the feedback. I updated the tests, which I think fixes things hopefully.
Btw, since the [initial](https://github.com/openvinotoolkit/openvino/pull/21459/commits/d651d7721da44bcae5ebb7d155a6a702aeb7f22e) commit had changes to extra unnecessary files, I updated that initial commit to remove those file changes, and then pushed the updated test changes in the [second](https://github.com/openvinotoolkit/openvino/pull/21459/commits/9e2ce484c74604012b8800b4bff6a54a0c5ce4e5) commit.

I am not sure if this is how it should be done. Please let me know if you have any feedback :) 

Best,
Ahmad

**Comment by rkazants** [comment]
build_jenkins

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend{
+namespace tensorflow{
+namespace op{
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:13]:
```suggestion
namespace frontend {
namespace tensorflow {
namespace op {
```

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+OutputVector translate_mul_no_nan_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"MulNoNan"});
+
+    // first = x, second = y
+    auto x = node.get_input(0);
+    auto y = node.get_input(1);
+
+    // prepare zero constants of the same type as the inputs
+    auto zero = make_shared<v0::Constant>(element::f32, Shape{}, 0.0f)->output(0);
+    zero = make_shared<v1::ConvertLike>(zero, y);
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:23]: please use routine from utils:

```
auto const_zero = create_same_type_const_scalar<int32_t>(x, 0);
```

Do not forget to include `utils.hpp`

**Review by rkazants — APPROVED** [review]
Looks good. Left one comment

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp (hunk 2 above)
Comment (ahmadchalhoub) [reply src/frontends/tensorflow_common/src/op/mul_no_nan.cpp]: @rkazants thanks for the review! Just updated the requested change.

**Comment by rkazants** [comment]
build_jenkins

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#include "common_op_table.hpp"
+#include "utils.hpp"
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:2]:
```suggestion
```

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#include "common_op_table.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:4]:
```suggestion
#include "utils.hpp"
```

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
+#include "common_op_table.hpp"
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:1]:
```suggestion
// Copyright (C) 2018-2023 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include "common_op_table.hpp"
```

File: src/frontends/tensorflow_common/src/op/mul_no_nan.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:5]:
```suggestion
#include "common_op_table.hpp"
#include "openvino/op/equal.hpp"
```

**Comment by rkazants** [comment]
build_jenkins

**Pull request merged**